pkg/initializer/initializer.go

```diff
@@ -60,22 +65,18 @@ func (e *EtcdInitializer) Initialize(mode validator.Mode, failBelowRevision int6
 		}
 		m := member.NewMemberControl(e.Config.EtcdConnectionConfig)
 		isScaleup, err := m.IsClusterScaledUp(ctx, clientSet)
-		if isScaleup && err == nil {
-			retry.OnError(retry.DefaultBackoff, errors.IsErrNotNil, func() error {
-				// Additional safety check before adding a learner
-				if _, err := os.Stat(e.Config.RestoreOptions.Config.DataDir); err == nil {
-					if err := os.RemoveAll(filepath.Join(e.Config.RestoreOptions.Config.DataDir)); err != nil {
-						return fmt.Errorf("failed to remove directory %s with err: %v", e.Config.RestoreOptions.Config.DataDir, err)
-					}
-				} else if !os.IsNotExist(err) {
-					return err
-				}
-				return m.AddMemberAsLearner(ctx)
-			})
-			// return here after adding non-voting member(learner) as no restoration or validation needed
-			return nil
-		} else if err != nil {
+		if err != nil {
 			logger.Errorf("scale-up not detected: %v", err)
+		} else if isScaleup && err == nil {
```

Comment:
in the else clause there is no need to check `err == nil` as it will always be true.

Reply:
yes, agreed